File: eTeatar/eTeatar.Model/Korisnik.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eTeatar.Model
{
    public class Korisnik
    {
        public int KorisnikId { get; set; }

        public string Ime { get; set; }

        public string Prezime { get; set; }

        public string Email { get; set; }

        public string Telefon { get; set; } 

        public string KorisnickoIme { get; set; }

        public byte[]? Slika { get; set; }

        public DateTime DatumRodenja { get; set; }

        public DateTime DatumRegistracije { get; set; }

        public virtual ICollection<Karta> Karta { get; set; } = new List<Karta>();

        public virtual ICollection<KorisnikUloga> KorisnikUlogas { get; set; } = new List<KorisnikUloga>();

        public virtual ICollection<Ocjena> Ocjenas { get; set; } = new List<Ocjena>();

        public virtual ICollection<Rezervacija> Rezervacijas { get; set; } = new List<Rezervacija>();

        public virtual ICollection<Uplata> Uplata { get; set; } = new List<Uplata>();

        public virtual ICollection<Vijest> Vijests { get; set; } = new List<Vijest>();
    }
}

File: eTeatar/UI/eteatar_desktop/lib/models/korisnik.dart
import 'package:eteatar_desktop/models/korisnik_uloga.dart';
import 'package:json_annotation/json_annotation.dart';

part 'korisnik.g.dart';

@JsonSerializable()
class Korisnik {
  int? korisnikId;
  String? ime;
  String? prezime;
  String? email;
  String? slika;
  String? telefon;
  String? korisnickoIme;
  DateTime? datumRodenja;
  String? lozinka;
  String? lozinkaPotvrda;
  List<KorisnikUloga> korisnikUlogas;

  Korisnik({
  this.korisnikId,
  this.ime,
  this.prezime,
  this.email,
  this.slika,
  this.telefon,
  this.korisnickoIme,
  this.datumRodenja,
  this.lozinka,
  this.lozinkaPotvrda,
  this.korisnikUlogas = const [],
});

  factory Korisnik.fromJson(Map<String, dynamic> json) => _$KorisnikFromJson(json);

  /// Connect the generated [_$GlumacToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$KorisnikToJson(this);
}